### PR TITLE
Fix blocked detectors

### PR DESCRIPTION
**Pull request by zguesmi**
- When detecting unnotified contribute/reveal use the initialization block instead of the current block to lookup for the contribute/reveal metadata.
- Set deadlines when adding new tasks.
- Don't detect for expired tasks

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -9,19 +9,20 @@ spring:
...
-    dealwatcherreplay.period: 60000       # 1 min
-    reveal.timeout.period: ${REVEAL_TIMEOUT_PERIOD:120000}       # 2 min
-    reveal.unnotified.period: 30000       # 30 s
-    contribution.unnotified.period: 30000 # 30 s
-    contribution.timeout.period: 120000   # 2 min
-    unstartedtx.period: 300000            # 5 min
-    task.initialized.unnotified.period: 30000  # 30 s
-    task.finalized.unnotified.period: 30000  # 30 s
-  stdoutAvailabilityDays: ${IEXEC_STDOUT_AVAILABILITY_DAYS:3} # 3 days after final deadline
+    worker-lost:      3000  # 30s
```
Comment (jeremyjams) [inline src/main/resources/application.yml:15]:
```suggestion
    worker-lost:      30000  # 30s
```

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -9,19 +9,20 @@ spring:
...
+    chain:
+      unstarted-tx:   300000  # 5m
+      initialize:     30000   # 30s
+      contribute:     3000   # 30s
```
Comment (jeremyjams) [inline src/main/resources/application.yml:19]:
```suggestion
      contribute:     30000   # 30s
```

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -9,19 +9,20 @@ spring:
...
+      reveal:         30000   # 30s
+      finalize:       30000   # 30s
+    timeout:
+      contribute:     1200  # 2m
```
Comment (jeremyjams) [inline src/main/resources/application.yml:23]:
```suggestion
      contribute:     120000  # 2m
```

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -75,4 +76,8 @@ management:
 
 graylog:
   host: ${IEXEC_CORE_GRAYLOG_HOST:localhost}
-  port: ${IEXEC_CORE_GRAYLOG_PORT:12201}
\ No newline at end of file
+  port: ${IEXEC_CORE_GRAYLOG_PORT:12201}
+
+stdout:
```
Comment (jeremyjams) [inline src/main/resources/application.yml:81]: I think we should keep days here & use `TimeUnit.DAYS.toMillis(purgeRate)`.

File: src/main/java/com/iexec/core/task/TaskService.java [hunk 5]
```diff
@@ -105,6 +109,8 @@ public TaskService(
                 .orElseGet(() -> {
                         Task newTask = new Task(chainDealId, taskIndex, imageName,
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/task/TaskService.java:110]: We could probably use a builder

File: src/main/java/com/iexec/core/task/TaskService.java [hunk 6]
```diff
@@ -187,7 +198,18 @@ public Date getTaskFinalDeadline(String chainTaskId) {
      */
     // TODO change this mechanism of update
     public CompletableFuture<Void> updateTask(String chainTaskId) {
-        long expiration = getTaskFinalDeadline(chainTaskId).getTime();
+        Optional<Task> oTask = getTaskByChainTaskId(chainTaskId);
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/task/TaskService.java:201]: :+1:

**Review by jeremyjams — APPROVED** [review]

File: src/main/resources/application.yml (hunk 4 above)
Comment (zguesmi) [reply src/main/resources/application.yml:81]: Done

File: src/main/java/com/iexec/core/task/TaskService.java (hunk 5 above)
Comment (zguesmi) [reply src/main/java/com/iexec/core/task/TaskService.java:110]: This needs modifications of the Task class (to generate the chainTaskId with the builder). So I'll leave as it is for now. We can change it in the future. Thanks anyway!

File: src/main/java/com/iexec/core/chain/IexecHubService.java [hunk 7]
```diff
@@ -125,7 +125,7 @@ public boolean isBeforeContributionDeadline(ChainDeal chainDeal) {
     }
 
     /**
-     * <p> Get a deal's contribution deadline date. The deadline
+     * <p> Get deal's contribution deadline date. The deadline
      * is calculated as follow:
      * start + maxCategoryTime * nbOfCategoryUnits.
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/chain/IexecHubService.java:130]:
```suggestion
     * start + maxCategoryTime * maxNbOfPeriods.
```

Comment (zguesmi) [reply src/main/java/com/iexec/core/chain/IexecHubService.java:130]: Merci !

File: src/main/java/com/iexec/core/detector/replicate/ContributionUnnotifiedDetector.java [hunk 8]
```diff
@@ -39,31 +39,36 @@
     private final ReplicateStatus offchainCompleting;
     private final ReplicateStatus offchainCompleted;
     private final ChainContributionStatus onchainCompleted;
-    private final CoreConfigurationService coreConfigurationService;
+    public final int detectorRate;
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/detector/replicate/ContributionUnnotifiedDetector.java:42]:
```suggestion
    private final int detectorRate;
```

File: src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java [hunk 9]
```diff
@@ -123,15 +125,15 @@ private void updateReplicateStatuses(String chainTaskId, Replicate replicate, Re
                 case CONTRIBUTED:
                     // retrieve the contribution block for that wallet
                     ChainReceipt contributedBlock = iexecHubService.getContributionBlock(chainTaskId,
-                            wallet, web3jService.getLatestBlockNumber());
+                            wallet, initBlocknumber);
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/detector/replicate/UnnotifiedAbstractDetector.java:128]: :clap:

**Review by jeremyjams — APPROVED** [review]

**Pull request merged**